assets/app/scripts/directives/util.js

```diff
@@ -75,11 +75,11 @@ angular.module('openshiftConsole')
         tag: '=?'
       },
       controller: function($scope, annotationFilter, iconClassFilter) {
-        var icon = $scope.icon = annotationFilter($scope.resource, $scope.tag ? $scope.tag + ".icon" : "icon");
+        var icon = $scope.icon = annotationFilter($scope.resource, "icon", $scope.tag);
         $scope.isDataIcon = icon && icon.indexOf("data:") == 0;
         if (!$scope.isDataIcon) {
           // The icon class filter will at worst return the default icon for the given kind
-          $scope.icon = iconClassFilter($scope.resource, $scope.kind, $scope.tag ? $scope.tag + ".iconClass" : "iconClass");
+          $scope.icon = iconClassFilter($scope.resource, $scope.kind, "iconClass", $scope.tag);
```

Comment:
I think this is the only place we're calling iconClassFilter... just roll the logic into this one function